endaq/batch/core.py

```diff
@@ -103,21 +107,51 @@ def _make_metrics(ch_data_cache: analyzer.CalcCache):
     - temperature - degrees Celsius
     - pressure - kiloPascals
     """
+    if include and exclude:
+        raise ValueError("parameters `include` and `exclude` are mutually-exclusive")
+
+    VALID_METRICS = {
```

Comment:
It seems like there should be a cleaner way to do this... what about just making the docstring the name, and then gathering them like:
``` python
VALID_METRICS = {v.__doc__.strip().casefold(): k for k,v in CalcCache.__dict__.items() if k.endswith("Full")}
```
You could have the docstring for each metric contain the name and the description with some sort of delimiter, like `"""RMS Velocity: Velocity Tri-axial RMS, after applying a 0.1Hz highpass filter."""` and do `v.__doc__.partition(':')[0].strip().casefold()`

Reply:
I appreciate the suggestion, but I'm not sure it would make the code "cleaner". Specifically, I don't like that a developer wouldn't know what metrics are included in `VALID_METRICS` without running the code (or, I suppose, tediously reading the method names for `CalcCache`). I would also prefer to avoid adding functional significance to the docstrings; if I had to I might instead add the display names as attributes of the method objects themselves, probably via some kind of decorator function.

Reply:
It keeps everything defined in one place (names, descriptions, and the cached properties themselves). It's less tedious than keeping two modules updated together. New metrics can be added by simply implementing new properties in `CalcCache`.

A decorator would be better than parsing the docstring, but I wasn't sure if the stacked decorators (one with parameters) wouldn't create complications.

Reply:
Unfortunately I don't think this is a productive conversation. Please do one of the following:

- if you've found a regression or a feature-breaking bug that would be introduced by this commit, please describe the bug and review the PR as "Requested changes".
- if you think the high-level feature (not regarding the implementation) is undesirable, please discuss your concerns and review the PR as "Requested changes"; we can later choose to close the PR.
- if you have questions about either the feature or the implementation that impedes your understanding of the code, please review the PR as "Commented" and ask the questions you have.
- otherwise, if you have disagreements about the implementation *but not the feature itself, and not concerning any undesirable behavior or bugs*, please review the PR as "Approved" **and describe your concerns in a GitHub issue on this repo** so that we can discuss it in the future without holding up new features.